### PR TITLE
Strict permission checking only upon request

### PR DESCRIPTION
**Pull request by pde**
Use --strict-permissions if you're running as a privileged user on a system
  where non-privileged users might have write permissions to parts of the lets
  encrypt config or logging heirarchy.  That should not normally be the case.

  Working toward a fix for #552

### CODE REVIEW

**Comment by pde** [comment]
OK, I think this is ready for review! @jdkasten ?

File: letsencrypt-nginx/letsencrypt_nginx/tests/configurator_test.py [hunk 1]
```diff
@@ -14,7 +14,6 @@
 
 from letsencrypt_nginx.tests import util
 
-
```
Comment (jdkasten) [inline letsencrypt-nginx/letsencrypt_nginx/tests/configurator_test.py]: FYI: Two blank lines before classes or class level functions is standard pep8.
https://www.python.org/dev/peps/pep-0008/#blank-lines

File: letsencrypt-nginx/letsencrypt_nginx/tests/util.py [hunk 2]
```diff
@@ -60,6 +61,13 @@ def get_nginx_configurator(
         name="nginx",
         version=version)
     config.prepare()
+    # also make a general client config for good measure...
+    namespace = mock.MagicMock(
```
Comment (jdkasten) [inline letsencrypt-nginx/letsencrypt_nginx/tests/util.py]: This namespace object should probably be the config created inline on line 51.  

You avoid differing config objects, which would create some confusion.  Given the testing suite already relies on some of these config variables.  (The config_dir parameter of this function is an existing directory and the tests rely on writing to it.)

File: letsencrypt/cli.py [hunk 3]
```diff
@@ -671,7 +675,7 @@ def create_parser(plugins, args):
 
 # For now unfortunately this constant just needs to match the code below;
 # there isn't an elegant way to autogenerate it in time.
-VERBS = ["run", "auth", "install", "revoke", "rollback", "config_changes",\
+VERBS = ["run", "auth", "install", "revoke", "rollback", "config_changes",
```
Comment (jdkasten) [inline letsencrypt/cli.py]: Nice, I had lined up to remove this in my next PR. :)

File: letsencrypt/crypto_util.py [hunk 4]
```diff
@@ -45,8 +44,12 @@ def init_save_key(key_size, key_dir, keyname="key-letsencrypt.pem"):
         logger.exception(err)
         raise err
 
+    import zope.component
+    from letsencrypt import interfaces
```
Comment (jdkasten) [inline letsencrypt/crypto_util.py]: Safe to move these imports to the top, also reduces duplicate code in lines 79-80

**Comment by jdkasten** [comment]
Great!  Thanks @pde for taking care of this long standing problem.  I can take care of the formatting issues if you don't mind.  In general, spacing has been deliberate following the google style guide.  https://google-styleguide.googlecode.com/svn/trunk/pyguide.html#Imports_formatting.

**Comment by jdkasten** [comment]
LGTM! Ran tests with and without --strict-permissions.  Changed permissions on a base directory and the error message was appropriate when run with strict permissions and ran just fine without it turned on.

**Comment by jdkasten** [comment]
@pde if you are fine with (what I believe to be) formatting changes.  I think it is good to merge.

**Comment by pde** [comment]
Great, thanks!

**Pull request merged**